### DISCUSSION TITLE
Web firmware upgrade stuck on switching to safeboot partition page- esp32s3 with w5500 SPI ethernet chip - bricks the device!

### DISCUSSION THREAD

**Discussion by nccchirag** [arendst/Tasmota#22261] [Q&A] [answered]
**PROBLEM DESCRIPTION**

Im using esp32s3 with w5500 ethernet chip [8 = ETH_PHY_W5500 (SPI)] and firmware upgrade via web does not work via ethernet .. the UI is stuck on "switching to safeboot" page .. thereafter i need to serially flash esp to restore.. however in the same setup if I enable wifi and then perform an upgrade it works fine!

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used ESP32S3 with generic W5500 USR-ES1
- [x] Tasmota binary firmware version number used: v14.2.0.6
- [ ] Pre-compiled
- [x] Self-compiled
- [x] Flashing tools used: esptool

**TO REPRODUCE**

1. custom compile latest dev build with USE_ETHERNET
2. set -> EthType 8;EthAddress 1;EthClockMode 3; Wifi 0;
3. navigate to http://ESP32S3_eth_ip/up?
4. select the .bin file and upload
5. observe the UI stuck on "switching to safeboot partition" with no log on serialport / console even with seriallog 4 settings

**EXPECTED BEHAVIOUR**

Local .bin file upgrade via web UI accessed from ethernet IP should work

**SCREENSHOTS**

**ADDITIONAL CONTEXT**

tried custom compilation with the below mentioned platform releases but no luck .. all have the same issue
1. 2024.06.10
2. 2024.07.11
3. 2024.08.10
4. 2024.09.30

requesting help for the above ... thank you

**Comment by nccchirag** [comment]
Safeboot stuck screenshot
![image](https://github.com/user-attachments/assets/c4860a18-454d-434b-8f11-9b689afc566f)

Serial Log
![image](https://github.com/user-attachments/assets/50f51842-82fb-4729-a24d-62f488622349)

**Comment by arendst** [comment]
My best guess is that the safeboot firmware just does not support ethernet; it's just not included in the binary because it takes up too much space.

So in short, as designed.

**Reply by nccchirag** [reply]
thank you for the prompt response

can I compile / upgrade without safeboot?

**Reply by nikito7** [reply, marked as answer]
```
[env:tasmota32s3eth-safeboot]
extends                 = env:tasmota32_base
board                   = esp32s3-qio_qspi
board_build.app_partition_name = safeboot
build_flags             = ${env:tasmota32_base.build_flags}
                         -DFIRMWARE_SAFEBOOT
                         -DUSE_SPI
                         -DUSE_ETHERNET
lib_extra_dirs          = lib/lib_ssl, lib/libesp32
lib_ignore              = ${safeboot_flags.lib_ignore}
```

You need flash it with esptool at correct address.

**Reply by sfromis** [reply]
You could also do the upgrade via Wifi, when you're already "in". Use the `OtaUrl` command if needed (value will not be saved), and `Upgrade 1`.

**Reply by nccchirag** [reply]
Thank you @nikito7 @sfromis 
Will try both suggestions and update tomorrow

**Reply by nccchirag** [reply]
> ```
> [env:tasmota32s3eth-safeboot]
> extends                 = env:tasmota32_base
> board                   = esp32s3-qio_qspi
> board_build.app_partition_name = safeboot
> build_flags             = ${env:tasmota32_base.build_flags}
>                          -DFIRMWARE_SAFEBOOT
>                          -DUSE_SPI
>                          -DUSE_ETHERNET
> lib_extra_dirs          = lib/lib_ssl, lib/libesp32
> lib_ignore              = ${safeboot_flags.lib_ignore}
> ```
> 
> You need flash it with esptool at correct address.

have successfully compiled using the modified tasmota_configurations_ESP32.h ... I think the default partition scheme used is esp32_partition_app2880k_fs320k.csv & the address mentioned for safeboot is 0x10000 ... 

> esptool --chip esp32s3 --port COM13 write_flash 0x10000 .\tasmota32s3eth-safeboot.bin

tas community rocks! I can confirm the proposed solution does work! 

thank you very much for the prompt support everyone 🙏🏻

**Comment by nikito7** [comment]
https://github.com/arendst/Tasmota/pull/21983

**Comment by nccchirag** [comment]
I also tried the below to avoid reflashing the custom safeboot file .. 
replaced the custom compiled safeboot file in the "variants" folder and then compiled the firmware and voila the factory bin works too!

![image](https://github.com/user-attachments/assets/33584899-30c1-4420-9da1-fe1ee7d2faeb)

**Reply by nikito7** [reply]
Avoid to use "Upgrade 2" without create your own ota server.

Upgrade 2 - updates safeboot

**Comment by Jason2866** [comment]
It can be even more automated. Copy the above safeboot env in `platformio_tasmota_cenv.ini` rename it to `[env:tasmota32s3-safeboot]` and compile this env before compiling a tasmota32-s3 firmware.
The modified s3 safeboot will be copied automatically in variants folder and will be used for all s3 firmwares!
